### DISCUSSION TITLE
Cloudlog as Clubstation Log and multi user operation

### DISCUSSION THREAD

**Discussion by Tonimara** [magicbug/Cloudlog#2348] [Q&A]
First thank you for this great project - I implemented the docker version (cluster recommendation) and will use it as my log. My question is how to configure users (operators and admin), logbook and locations in order to use it for our clubstation with operators being active in parallel from one location? Thank you for any hints and experiences for that use case.

**Comment by DG7YEO** [comment]
Same question here, with the 1.x Versions it was no problem to create additional users and they could use the Log. When I now create a new user or login with the profile of an old test user I see the message "Attention: you need to set an active station location." and the Station Logbooks/Location pages are empty. 

Is there any way to make the other users able to see those logs and enter there QSOs? Because this was the main reason why I moved with the clubstation Log to a cloudlog setup and with the last 1.x Version I had this was possible.

vy73
Elmar
DG7YEO

**Reply by int2001** [reply]
At the moment this is not possible. Only workaround: share login. That works

**Reply by DG7YEO** [reply]
Hm,
it's a pity that one of the main features why I choose cloudlog as Log Software for our club calls get lost with an upgrade to a higher version.

**Reply by int2001** [reply]
ehh. that "feature" is not available since station_location-Branch.
and that's a long time ago.

Sorry. But we have this on some kind of roadmap...

**Reply by DG7YEO** [reply]
Only I can say it worked for us even with the station_location, it stopped working when I updated to a version using Station Logbook. This was in Spring 2023 I moved from 1.9 to 2.3. Before it was working. 
Anyway it is like it is. Cloudlog is still a great log software, even if I have to share my login now with the other users and need to touch most QSO a second time to have the correct callsing in the Station Operator field. 

73
DG7YEO

**Reply by int2001** [reply]
so maybe downgrade is an option for you?

The operator-field is a good idea in between. Let me add that to the QSO-Form, so you only have to type it in once (at the first QSO)?

May that help?

**Reply by DG7YEO** [reply]
Yes, this would definitely help.

**Reply by int2001** [reply]
Once this (https://github.com/magicbug/Cloudlog/pull/2421) is approved/merged by @magicbug it is available.
You enter the Op-Call once and it stays there until you log out.
Multiple-Sessions with the same login and different Op-Calls are possible.

Screenshot from Add-QSO-Form:

![Screenshot of AddQSO-Form](https://github.com/magicbug/Cloudlog/assets/1410708/84de2a6f-a7d3-4029-a9a3-daf2de973c94)

**Comment by int2001** [comment]
The Operator-Feature is now in the dev-branch.
But for Real-Multi-User i got a bunch of questions.

At first: Cloudlog is a Multiuser-Application. Means: Users with own logins can manage their QSOs within their Logbooks and Station-Locations. A "shareable" Logbook has to be designed well, because it breaks with the "isolated logbooks" for a reason.

i think we should discuss the requested feature(s) together here, before we begin to implement something with doesn't find the path into the Applications. 

some guidance/consens is needed. (i am looking to @magicbug as well as @AndreasK79 and @phl0)

- should the multiop-user be able to do all the things with the station_id like the owner? (Delete, edit, credentials for 3rd party-services, …)
- if not, where’s the cut?
- could the multiop-user set the location as “active location”, or only when adding qsos?
- what about displaying past QSOs (dash, overview, logbook-advanced)?
- what about awards/analytics and so on? Full access to the MultiOp-User?

Should we only grant some functions to an admin? 
- Who is the admin? 
- The Main-Owner of the Call? 
- The Instance-Admin?

Multi-Op-User: The user which receives at least some grants to add QSOs to a Station-Location which he doesn't own in the classic sense

you see: shareable Logs are not so easy to design. Any input is helpful, before one starts coding.

**Reply by DG7YEO** [reply]
Hi,
here my view as a user on this.

Currently there are 2 user profiles may match, Administrator and Editor. If you stay with these for shared Logs only the Admin should be able to create them as well as the station locations and which other exiting users are allowed to use this shared log. The Editor should only be able to view all QSOs, as well the once from other operators, but enter and edit only the own QSOs. Delete should be limited to the Admin or at least need a confirmation from the admin.

Other option might be to using Groups, so the shared Logbook is assigned to a group there is at least one group admin who can do all log related tasks like create locations, eQSL stuff ... while all other members of that group have rights like described for the Editors above. This might open the option that a group of hams already share this log for a club call can easily assigned an other log i.e. a special event station activated by the same group.

I'm not a software engineer, so my view might not the best way in case of how to code such a feature.

**Reply by huberse** [reply]
Hi

same here for our club. Thats actually why i still have a v1.x installed on our own web instance as we completely operate the station from remote and have the log in the cloud. This served us well since years and still works like a charm https://www.hb9lf.ch/Cloudlog  My problem will start end of April when my hoster decomissions PHP 7. I already tried with v2.x but had as well the issue that i can't assign logins to the club log. See #2007

Needed Roles : Administrator and Editor. 

The Administrator (QSL Manager) should be able to create Stations, Locations and logbooks. He then should be able to create and assign the Club Members (Operators/Editors) to the logbooks so that they can use their own login. The transfer to other logbooks (eQSL, lotw) should also be restricted to the Administrator Role.

The Editors should only be able to create, search and edit their own QSO's. If they create a QSO, theire Callsign should be taken into the Operator field. They should also be able to use the API (CloudlogCAT, adifpush).

thanks for looking into this
73' Serge   HB9EDI

**Reply by HB9HIL** [reply]
Hello Serge

We have this on our roadmap. But wen can't give you an ETA and obviously we can't promise that this will happen until April 2024...

Vy 73 de HB9HIL

Liebe Grüsse aus Appenzell

**Reply by huberse** [reply]
Hi Fabian,

thanks for the reply and info on the roadmap. I just found out thta i can extend php 7 till september ... In case i can support something, let me know. Some php skills are left or do whatever testing is needed.

Vy 73' Serge  HB9EDI       Grüsse aus dem Laufental
